### PR TITLE
Nightlight

### PR DESCRIPTION
**Pull request by Geevies**
Opening PR so someone smarter than me can figure out why spawning doesn't turn the light on in the area you spawn in.

Inspired by https://github.com/Baystation12/Baystation12/pull/19585, thanks chimpsky you legend

### CODE REVIEW

File: code/game/area/areas.dm [hunk 1]
```diff
@@ -105,6 +105,18 @@
 
 	. = ..()
 
+/mob/living/carbon/human/Initialize()
+	. = ..()
+	world << round_start_time
+	world << world.time
+	var/area/A = get_area(src)
+	if(A?.lightswitch == FALSE)
```
Comment (Arrow768) [inline code/game/area/areas.dm:113]: ?. is unneeded as the lightswitch var is defined on /area

File: code/game/area/areas.dm [hunk 2]
```diff
@@ -105,6 +105,18 @@
 
 	. = ..()
 
+/mob/living/carbon/human/Initialize()
+	. = ..()
+	world << round_start_time
+	world << world.time
```
Comment (Arrow768) [inline code/game/area/areas.dm:111]: Leftover debug lines

**Comment by Geevies** [comment]
!review Works, thanks to Arrow's code and Skull's pointing out on how job spawning works.

File: code/controllers/subsystems/job.dm [hunk 3]
```diff
@@ -341,6 +341,8 @@
 		if(istype(S, /obj/effect/landmark/start) && istype(S.loc, /turf))
 			H.forceMove(S.loc)
 			H.lastarea = get_area(H.loc)
+			H.lastarea.lightswitch = TRUE
+			H.lastarea.set_lightswitch(TRUE)
```
Comment (skull132) [inline code/controllers/subsystems/job.dm:345]: This is just, really bad design. If you have a function that's called `set_x` which accepts a new value, then by god that function should actually set the variable it's describing.

So basically: the only thing you should need in this line is `H.lastarea.set_lightswitch(TRUE)`.

File: code/game/area/areas.dm [hunk 4]
```diff
@@ -105,6 +105,12 @@
 
 	. = ..()
 
+/area/proc/set_lightswitch(var/state)
+	var/obj/machinery/light_switch/L = locate() in src
```
Comment (skull132) [inline code/game/area/areas.dm:109]: As explained above, this should also set the lightswitch variable. There is no good reason for the light switches to be out of sync with the area's light state.
```suggestion
	lightswitch = state
	var/obj/machinery/light_switch/L = locate() in src
```

File: code/controllers/subsystems/job.dm [hunk 5]
```diff
@@ -341,6 +341,8 @@
 		if(istype(S, /obj/effect/landmark/start) && istype(S.loc, /turf))
 			H.forceMove(S.loc)
 			H.lastarea = get_area(H.loc)
+			H.lastarea.lightswitch = TRUE
```
Comment (Geevies) [inline code/controllers/subsystems/job.dm:344]:
```suggestion
```

File: code/controllers/subsystems/job.dm (hunk 3 above)
Comment (Geevies) [reply code/controllers/subsystems/job.dm]: i have no idea why i did this

**Pull request merged**